# 20201107 N156W cloning

## 20201107 OE-PCR
Overlap extension PCR for N156W, as quikchange didn't work

- primers

Name | sequence | Length | Tm (NEB for Q5) | % GC |
---- | -------- | ------ | --------------- | ---- |  
N156W_OE-PCR_F | TGGTACAACTTTGAAAAGCCTTTC | 24 | 63 | 38 | 
N156W_OE-PCR_R | CCAGGACTTGGCAGAAATATCGTA | 24 | 66 | 46 | 

    ```
    <-----ACC
        TGG----->
    ```

- used NEB's Tm calculator for Q5, said to anneal at 64C
- Ran a PCR:
    Reagent          | Volume (uL) 
    ---------------- | ------
    2x Q5 Master Mix | 25
    template         | 1
    10 uM primers    | 1
    H2O              | 23
    **TOTAL**        | **50**

- PCR Protocol:
    - 98C, 5 min
    - Repeat 18x:
        - 98C, 30 sec
        - 64C, 30 sec
        - 72C, 7 min
    - 72C, 10 min

## 20201108 Ligation and transformation

### Dpn1 digestion
- added 1 uL to 50 uL PCR, incubated 1 hr at 37C
- PCR cleanup using 5x volume = 250 uL PB, applied to purple column, washed twice in 750 uL PE, eluted with 50 uL EB
- nanodrop: 13.8 ng/uL
- estimated amount of DNA = 13.8 ng/uL * 50 uL * (1 pmol / (6026 bp * .66 ng per bp) = .1735 pmol

### T4 PNK phosphorylation and T4 Ligase ligation
- [PNK protocol](https://www.neb.com/protocols/0001/01/01/non-radioactive-phosphorylation-with-t4-pnk-or-pnk3-phosphatase-minus) says up to 300 pmol of DNA, so we're far from saturating the reaction
- Using T4 DNA Ligase Buffer instead of T4 PNK Reaction Buffer because the 10x ligase buffer already has 10 mM ATP, so don't need to prepare ATP separately (this is noted as an option in the protocol, as T4 PNK exhibits 100% activity in the ligase buffer)
    reagent | vol (uL)
    ------- | --------
    DNA | 40
    10X T4 DNA Ligase Buffer | 5
    T4 PNK | 1
    Water | 4
    **Total** | **50**
- incubated 35 min at 37C followed by heat inactivation for 20 min at 65C, cooled to RT
- added 2.5 uL of T4 DNA Ligase to the 50 uL reaction, 10 min at RT, then heat inactivation for 10 min at 65C (see [protocol](https://www.neb.com/protocols/0001/01/01/dna-ligation-with-t4-dna-ligase-m0202))
- transformed 1 uL into Top10 cells

## 2020-11-09
- Transformation worked, got several colonies.
- Ran colony PCR on six colonies, with oCJM0047
- PCR recipe:
    Reagent          | Volume (uL) 
    ---------------- | ------
    2x Master Mix    | 10
    template         | 1
    10 primers       | 1
    H2O              | 8
    **TOTAL**        | **20**
- PCR Protocol:
    - 98C, 3 min
    - Repeat 35x:
        - 98C, 30 sec
        - 52, 30 sec
        - 72C, 1 min
    - 72C, 2 min
- Also inoculated those same six colonies, 5mL LB+Kan

## 2020-11-10 Sequencing Results
- Results from sequencing: all of the 6 sequences had two W's (so TGGTGG) inserted instead of just one. Anum thinks that we should have put homology on both sides of the mutation, so:

    ```
    <-----ACC---
       ---TGG----->
    ```

- miniprepped colony A, one of the plasmids that is correct except for the extra TGG
- Will delete the extra TGG with quikchange primers:
    Name | sequence | Length | Tm (quikchange) | % GC |
    ---- | -------- | ------ | --------------- | ---- |  
    N156W_del_F | GATATTTCTGCCAAGTCCTGGTACAACTTTGAAAAGCC | 38 | 81 | 42 | 
    N156W_del_R | GGCTTTTCAAAGTTGTACCAGGACTTGGCAGAAATATC | 38 | 81 | 42 | 

    - Tm is calculated with the [quikchange formula provided by Agilent](https://www.chem-agilent.com/pdf/strata/200518.pdf): Tm = 81.5+0.41*42-675/38 = 81. Tm needs to be >= 78C so this is good

## 2020-11-12 Quikchange
- primers came
- Quikchange PCR mix:
  Reagent           | Volume (uL) 
  ----------------- | ------
  2x Master Mix     | 25
  1:20 dil template | 1
  10 uM primers     | 1
  H2O               | 23
  **TOTAL**         | **50**
- Quikchange PCR Protocol:
  - 98C, 10 min
  - Repeat 18x:
    - 98C, 30 sec
    - 55, 30 sec
    - 72C, 6 min
  - 72C, 10 min
- digested with Dpn1 for 1 hour at 37C
- transformed into Top10

## 2020-11-13 Colony PCR 
- Got 6 colonies
- Touched each and re-suspended in 25 uL water
- Used 5 uL to inoculate 5 mL LB+Kan cultures
- boiled 15 min at 95C
- used 1 uL (forgot to spin down cells) for PCR
- used primers oCJM0046/47

### PCR recipe:
Reagent          | Volume (uL) 
---------------- | ------
2x Master Mix    | 10
template         | 1
10 uM primers    | 1
H2O              | 8
**TOTAL**        | **20**

### PCR Protocol:
- 98C, 3 min
- Repeat 32x: (didn't have enough time for 35x)
  - 98C, 30 sec
  - 52C, 30 sec
  - 72C, 1 min
- 72C, 2 min

## 2020-11-14 Sequencing Results

- 4 reactions failed, 2 had pretty bad signal, 1 of those 2 looks like it has some weird insertions after the N156W codon.
- miniprepped the overnights to send plasmids for sequencing, and also re-did the same quikchange and transformation from 2020-11-12. I had rushed the transformation on 11/12 so perhaps that was the issue, not the primers.
- scaled up on 11/15
- miniprepped on 11/16
- sent for sequencing: 6 colonies from 11/14 minipreps, 6 colonies from 11/16

## 2020-11-15 Results
- 11/14 correct plasmids: C and F
- 11/16 correct plasmids: A, B, D, E, F

- Since the second transformation was cleaner, will use 11/16 A as the final version.