
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>2020-07-01 Gsp1 DMS Plan &#8212; GAP and GEF Expression Tuning</title>
    
  <link rel="stylesheet" href="../_static/css/index.73d71520a4ca3b99cfee5594769eaaae.css">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.40e2e510f6b7d1648584402491bb10fe.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/custom.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.3da636dd464baa7582d2.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.d31b09fe5c1d09cb49b26a786de4a05d.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Gsp1 sequence info" href="gsp1_sequence.html" />
    <link rel="prev" title="Meeting notes for DMS project" href="meeting_notes.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />



  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../index.html">
  
  <img src="../_static/logo.png" class="logo" alt="logo">
  
  
  <h1 class="site-logo" id="site-title">GAP and GEF Expression Tuning</h1>
  
</a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <ul class="current nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../journal/index.html">
   Experimental Journal
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../journal/20201204_Mutant_Expression.html">
     2020-12-04 Gsp1 Mutant Expression
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../journal/20201107_N156W_cloning.html">
     20201107 N156W cloning
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../journal/20201026_Mutant_Expression.html">
     2020-10-26 Gsp1 Mutant Expression
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../journal/2020-11-13%20HPLC%20and%20CD.html">
     2020-11-13 HPLC and CD with F28V, F28Y
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../journal/20201113_aggregation_test.html">
     2020-11-13 Aggregation Test
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../journal/20201029_Mutant_Expression.html">
     2020-10-29 Gsp1 Mutant Expression
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../journal/20201205_prep.html">
     20201205 through 20200110 - Gsp1 Mutant Expression
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../analysis/index.html">
   Analysis
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../analysis/20200112_plot_CD_HPLC.html">
     20200112_plot_CD_HPLC
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../GAP_assays/index.html">
   GAP Assays
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../GAP_assays/tina_notes.html">
     Notes from Tina about GAP assays
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../GAP_assays/20210121_sensor_calibration.html">
     2021-01-21 Phosphate Sensor Calibration
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../GAP_assays/20200122_sensor_calib.html">
     2020-01-22 Phosphate Sensor Calibration
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../GAP_assays/20210118/20210118_GAP_Assay.html">
     2021-01-18 GAP Assay
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../GAP_assays/20210117/20210117_GAP_Assay.html">
     2021-01-17 GAP Assay
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../GAP_assays/20210119/20210120_GAP_Assay.html">
     2021-01-20 GAP Assay
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../GAP_assays/20201117/20201117_GAP_assay.html">
     2020-11-17 GAP Assay
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="index.html">
   Notes
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="isntalling%20bio3d.html">
     2020-02-04 Installing bio3d
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="Neuwald2003_notes.html">
     Neuwald 2003 notes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="meeting_notes.html">
     Meeting notes for DMS project
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     2020-07-01 Gsp1 DMS Plan
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="gsp1_sequence.html">
     Gsp1 sequence info
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="20201021_results_outline.html">
     very rough results outline to consider
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="Gsp1%20annotation.html">
     2020-03-12 Gsp1 annotation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="Gsp1%20DMS%20Plan%20progress.html">
     2020-07-09 Gsp1 DMS Plan PROGRESS
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="For%20David%20%20What%20we%20know%20about%20mutants.html">
     2020-06-09 Mutation info for David
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="emap%20colony%20size%20notes.html">
     2020-03-27 EMAP Colony Size Notes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="Tina%20explaining%20the%20Gsp1%20EMAP%20paper%20SASA.html">
     2020-06-08 Tina on EMAP SASA
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="loose_ends.html">
     Loose ends - some general things to think about for DMS
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../protocols/index.html">
   Protocols
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../protocols/GAP_assay.html">
     GAP Assay
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="row topbar fixed-top container-xl">
    <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show">
    </div>
    <div class="col pl-2 topbar-main">
        
        <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
            data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
            aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
            title="Toggle navigation" data-toggle="tooltip" data-placement="left">
            <i class="fas fa-bars"></i>
            <i class="fas fa-arrow-left"></i>
            <i class="fas fa-arrow-up"></i>
        </button>
        
        
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../_sources/notes/Gsp1 DMS Plan 2020-07-01.md"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.md</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

        <!-- Source interaction buttons -->


        <!-- Full screen (wrap in <a> to have style consistency -->
        <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                title="Fullscreen mode"><i
                    class="fas fa-expand"></i></button></a>

        <!-- Launch buttons -->

    </div>

    <!-- Table of contents -->
    <div class="d-none d-md-block col-md-2 bd-toc show">
        
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i> Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#priorities">
   PRIORITIES
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#extra-sections">
   EXTRA SECTIONS
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#rosetta-ddg">
   ROSETTA DDG
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#sasa-computation">
   SASA COMPUTATION
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#comparison-histograms">
   COMPARISON HISTOGRAMS
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#gsp1-introduction-writing">
   GSP1 INTRODUCTION WRITING
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#allosteric-lobe-section">
   ALLOSTERIC LOBE SECTION
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#gtpase-conservation">
   GTPASE CONSERVATION
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#ras-dms-comparison">
   RAS DMS COMPARISON
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#dominant-negative-analysis">
   DOMINANT NEGATIVE ANALYSIS
  </a>
 </li>
</ul>

        </nav>
        
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="gsp1-dms-plan">
<h1>2020-07-01 Gsp1 DMS Plan<a class="headerlink" href="#gsp1-dms-plan" title="Permalink to this headline">¶</a></h1>
<div class="section" id="priorities">
<h2>PRIORITIES<a class="headerlink" href="#priorities" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><p>Rosetta ddG</p></li>
<li><p>SASA Computation</p></li>
<li><p>Comparison histograms</p></li>
<li><p>Gsp1 introduction writing</p></li>
</ul>
</div>
<div class="section" id="extra-sections">
<h2>EXTRA SECTIONS<a class="headerlink" href="#extra-sections" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><p>Allosteric lobe section</p></li>
<li><p>GTPase conservation</p></li>
<li><p>Ras DMS comparison</p></li>
<li><p>Dominant negative analysis</p></li>
</ul>
</div>
<div class="section" id="rosetta-ddg">
<h2>ROSETTA DDG<a class="headerlink" href="#rosetta-ddg" title="Permalink to this headline">¶</a></h2>
<ol class="simple">
<li><p>Compile benchmark datasets</p>
<ul class="simple">
<li><p>Download Kellogg and Protherm datasets from the kortemme lab website</p></li>
<li><p>Note the best score function (talaris 2013 for the Kellogg dataset), as those results should be used in analysis</p></li>
<li><p>Get Brandon’s dataset. We already have Hahnboem’s dataset (he emailed it)</p></li>
<li><p>Write code that will read in each dataset and have the list of mutants to include so that we can compute correlations based on dataset subsets and compare our results with published results (that use those subsets)</p></li>
<li><p>Get mutfiles from Shane and Kyle’s protocol capture</p></li>
</ul>
</li>
<li><p>Run cartesian ddG on all structures in benchmark</p>
<ul class="simple">
<li><p>Rosetta version: use master after Frank and Brandon’s recent commits fixing the relax repacking behavior are merged</p></li>
<li><p>Use the flags from Park et al 2016 JCTC for the preminimization and the ddG run, except -beta:cart should NOT be included because it is standard now (according to Hahnboem)</p></li>
<li><p>Pre-minimize all structures that appear in any dataset</p></li>
<li><p>Run ddG with all structures</p></li>
</ul>
</li>
<li><p>Analyze <code class="docutils literal notranslate"><span class="pre">cartesian_ddg</span></code> benchmark</p>
<ul class="simple">
<li><p>Compute correlation with experimental values for the full dataset, and for all of the subsets necessary to compare performance with published runs</p></li>
<li><p>Can also do the classifier task (destabilizing if ddg &lt; -1, stabilizing if ddg &gt; 1)</p></li>
<li><p>If performance is good (&gt;0.6 correlation, and hopefully at least as good as Park et al 2016), then use the protocol on Gsp1</p></li>
<li><p>If performance is not good, ask Hahnboem again for the code version used in Park et al 2016, and troubleshoot on that exact replicate run</p></li>
</ul>
</li>
<li><p>Run <code class="docutils literal notranslate"><span class="pre">cartesian_ddg</span></code> on Gsp1 (3m1i, residues 10-183)</p>
<ul class="simple">
<li><p>preminimize, check to make sure side chains are not repacked (or if they are, that behavior is expected)</p></li>
<li><p>make sure WT and MUT structures are written out for each run, so we can compare WT noise from run and look for sources of noise in MUT calculations</p></li>
<li><p>run full saturation mutagenesis</p></li>
</ul>
</li>
<li><p>Simple Ala scan (<code class="docutils literal notranslate"><span class="pre">mutate_residue</span></code> + minimize)</p>
<ul class="simple">
<li><p>Write the method (RosettaScripts or pyrosetta)</p></li>
<li><p>Run this method on benchmark set (preminimized with <code class="docutils literal notranslate"><span class="pre">cartesian_ddg</span></code>)</p></li>
<li><p>Compute correlation with experimental data, compare to <code class="docutils literal notranslate"><span class="pre">cartesian_ddg</span></code> results - can we use the simple Ala scan method for our purposes?</p></li>
<li><p>Run method on Gsp1</p></li>
</ul>
</li>
<li><p>Analysis:</p>
<ul class="simple">
<li><p>Compare Gsp1 results from 3 methods: <code class="docutils literal notranslate"><span class="pre">cartesian_ddg</span></code>, simple Ala scan, foldX</p></li>
<li><p>Correlate each of three methods with experimental EMPIRIC data</p></li>
<li><p>Can compute SASA vs. fitness and SASA vs. ddG metrics to see if the ddG matches SASA, which we expect it too</p></li>
<li><p>May try excluding residues from analysis if in van der waals contact with nucleotide (likely to have deleterious effects)</p></li>
<li><p>A good quality check is all heavy atom RMSD for the three lowest energy structures of each mutant (as ddG is calculated as mean for these 3 structures)</p></li>
<li><p>Predictability of experimental data:</p></li>
</ul>
</li>
<li><p>Extra ddG things we can do:</p>
<ul class="simple">
<li><p>compare yeast-to-human mutations against the whole distribution to inform if we can use human structures for ddG</p></li>
<li><p>can also just exclude residues from analysis if in van der waals contact with non-conserved residues</p></li>
<li><p>if we use human structures: run <code class="docutils literal notranslate"><span class="pre">cartesian_dd</span></code>, simple Ala scan, and/or <code class="docutils literal notranslate"><span class="pre">flex_ddg</span></code> on:</p>
<ul>
<li><p>open and closed monomer structures</p></li>
<li><p>GAP and GEF structures</p></li>
<li><p>All complexes</p></li>
</ul>
</li>
<li><p>Can run ddG methods on ubiquitin, and see how predictive it is for the experimental data, as an extra confirmation of the method being appropriate for Gsp1 analysis</p></li>
</ul>
</li>
</ol>
</div>
<div class="section" id="sasa-computation">
<h2>SASA COMPUTATION<a class="headerlink" href="#sasa-computation" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><p>Recompute SASA (burial) for the structures in PyRosetta or RosettaScripts</p></li>
<li><p>Recompute interface SASA for the interface residues</p>
<ul>
<li><p>“NONE” needs to be split up into “protein core” and “surface (non-interface)”. the latter may be relatively small n</p></li>
<li><p>definitely combine rim/support, consider combining with core</p></li>
</ul>
</li>
<li><p>Send David dataframe of interface positions and burial</p></li>
<li><p>Make a scatterplot of distance from nucleotide vs. fitness and compare to SASA (burial) vs. fitness</p></li>
</ul>
</div>
<div class="section" id="comparison-histograms">
<h2>COMPARISON HISTOGRAMS<a class="headerlink" href="#comparison-histograms" title="Permalink to this headline">¶</a></h2>
<ol class="simple">
<li><p>Make histograms with new binnings so we can look at them all together</p></li>
<li><p>Make stacked barplots, where “in van der waals contact with nucleotide” is shaded</p></li>
<li><p>Comparisons:</p>
<ul class="simple">
<li><p>Functional loops are enriched compared to other loops</p>
<ul>
<li><p>Do we need helices vs. loops vs. sheets plot, perhaps distinguishing non-functional vs. functional loops, and core vs. surface</p></li>
</ul>
</li>
<li><p>C-terminal linker with C-terminal helix</p>
<ul>
<li><p>There is a deleterious proline in the helix, is it an n1 proline? Highlight this.</p></li>
</ul>
</li>
<li><p>Protein core vs. surface exposed (non-interface) vs. surface exposed (interface)</p>
<ul>
<li><p>We expect interface core to be less important for Gsp1 than one would think for protein interaction interfaces in general</p></li>
</ul>
</li>
<li><p>Effector lobe vs. allosteric lobe</p>
<ul>
<li><p>note that the new binning loses many beneficial mutants</p></li>
<li><p>is effector lobe being worse only due t functional loops? Compare helices vs. sheets vs. func loops vs. non-func loops between lobes</p></li>
<li><p>what can we really say about this? talk to tanja</p></li>
</ul>
</li>
</ul>
</li>
<li><p>Come up with a single metric to summarize interface conclusions across many partners. Also could use this metric for the other comparison categories.</p>
<ul class="simple">
<li><p>One metric for interfaces could be number drop-outs in core of interface divided by number of drop-outs in interface (so including rim and support)</p></li>
</ul>
</li>
<li><p>Make a GEF interface plot:</p>
<ul class="simple">
<li><p>Stacked barplot showing GEF interface residues with those residues shared with GAP shaded</p></li>
<li><p>other categories are all other surface residues in an interface, all other surface residues not in an interface, and protein core residues</p></li>
<li><p>can consider doing this for other partners if it makes sense</p></li>
</ul>
</li>
</ol>
</div>
<div class="section" id="gsp1-introduction-writing">
<h2>GSP1 INTRODUCTION WRITING<a class="headerlink" href="#gsp1-introduction-writing" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><p>What Gsp1 does</p></li>
<li><p>Difficulty of studying Gsp1 as an essential gene</p></li>
<li><p>Studying point mutations to understand function, cite our paper</p></li>
</ul>
</div>
<div class="section" id="allosteric-lobe-section">
<h2>ALLOSTERIC LOBE SECTION<a class="headerlink" href="#allosteric-lobe-section" title="Permalink to this headline">¶</a></h2>
<p>If we include this section:</p>
<ul class="simple">
<li><p>For the structure figure, show side chains of beneficial mutations, looking like the contiguous blobs from Rama Ranganathan’s lab’s papers</p></li>
<li><p>Look at conservation of allosteric lobe ineractions</p>
<ul>
<li><p>align GEFs of all GTPases, perhaps Carla Mattos has a paper on this</p></li>
<li><p>Look at structural variation of the lobe across all GTPases</p></li>
</ul>
</li>
</ul>
</div>
<div class="section" id="gtpase-conservation">
<h2>GTPASE CONSERVATION<a class="headerlink" href="#gtpase-conservation" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><p>Look into Alfonso Valencia’s papers</p></li>
<li><p>How conserved is S. cerevisiae Gsp1 compared to other yeast homologs?</p></li>
<li><p>How conserved are other S.c. GTPases compared to their other yeast homologs?</p></li>
<li><p>How conserved is S.c. Gsp1 compared to other S.c. GTPases?</p></li>
</ul>
</div>
<div class="section" id="ras-dms-comparison">
<h2>RAS DMS COMPARISON<a class="headerlink" href="#ras-dms-comparison" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><p>Run structural alignment, to find corresponding residues between HRas and Gsp1</p>
<ul>
<li><p>which positions behave most similarly or most differently between the two experiments?</p></li>
</ul>
</li>
<li><p>NOTE: Tina says the Kuriyan lab is currently re-doing these experiments, as the deletion of the last half of alpha helix 5 (6?), as was done in Pradeep’s paper, seems to have destabilized the allosteric lobe mutants that were screened as deleterious (suggesting they may be false positives)</p></li>
</ul>
</div>
<div class="section" id="dominant-negative-analysis">
<h2>DOMINANT NEGATIVE ANALYSIS<a class="headerlink" href="#dominant-negative-analysis" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><p>Can look into gene dosage datasets from Weissman and Kirschner labs</p></li>
<li><p>Is Gsp1 conserved because mutants are dominant negative with WT? Look at Ran T24N and Ran Q71L cell biology papers</p></li>
</ul>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./notes"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        </div>
    </div>
    
    
    <div class='prev-next-bottom'>
        
    <a class='left-prev' id="prev-link" href="meeting_notes.html" title="previous page">Meeting notes for DMS project</a>
    <a class='right-next' id="next-link" href="gsp1_sequence.html" title="next page">Gsp1 sequence info</a>

    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Chris Mathy<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="../_static/js/index.3da636dd464baa7582d2.js"></script>


    
  </body>
</html>